include/app_memory/app_memdomain.h

```diff
@@ -82,32 +82,52 @@ struct app_region {
 #endif	/* CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT */
 
 #define appmem_partition(name) \
-	extern char *data_smem_##name; \
-	extern char *data_smem_##name##b; \
-	smem_size_declare(name);		  \
 	_app_dmem_pad(name) char name##_dmem_pad; \
 	_app_bmem_pad(name) char name##_bmem_pad; \
-	__kernel struct k_mem_partition mem_domain_##name; \
-	__kernel struct app_region name; \
+	__kernel struct k_mem_partition mem_partition_##name; \
+	__kernel struct app_region name;
+
+#define appmem_partition_header(name) \
+	extern char *data_smem_##name;	  \
+	extern char *data_smem_##name##b; \
+	smem_size_declare(name); \
+	extern struct k_mem_partition mem_partition_##name; \
+	extern struct app_region name; \
 	static inline void appmem_init_part_##name(void) \
 	{ \
 		name.dmem_start = (char *)&data_smem_##name; \
 		name.bmem_start = (char *)&data_smem_##name##b; \
 		smem_size_assign(name);				\
 		sys_dlist_append(&app_mem_list, &name.lnode); \
-		mem_domain_##name.start = (u32_t) name.dmem_start; \
-		mem_domain_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
-		name.partition = &mem_domain_##name; \
+		mem_partition_##name.start = (u32_t) name.dmem_start; \
+		mem_partition_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
+		name.partition = &mem_partition_##name; \
 	}
 
+/* A helper function that should be called to define multiple partitions*/
+#define APPMEM_PARTITION_OBJECT_DEFINE(...)\
+	FOR_EACH(appmem_partition, __VA_ARGS__)
+
+/* This defines all the functions that are needed to configure
+ * the memory partitions. Usually goes in a common header file.
+ */
+#define APPMEM_PARTITION_HEADER_DEFINE(...)\
+	FOR_EACH(appmem_partition_header, __VA_ARGS__)
+
+/* Returns the mem_partition struct */
+#define APPMEM_PARTITION_GET(name) (mem_partition_##name)
```

Comment:
Should this return the pointer?

Reply:
I could not find where this is used, actually

Reply:
> Should this return the pointer?

I guess you are right we should return a pointer.

>I could not find where this is used, actually

i will be pushing a new PR soon with a few test cases that have been ported to use these APIs.

Reply:
were you going to make this change?